// @capacitor/electron/dist/esm/electron/splashscreen.js
import * as tslib_1 from "tslib";
import { WebPlugin } from "@capacitor/core";
var SplashScreenPluginElectron = /** @class */ (function (_super) {
    tslib_1.__extends(SplashScreenPluginElectron, _super);
    function SplashScreenPluginElectron() {
        var _this = _super.call(this, {
            name: 'SplashScreen',
            platforms: ['electron']
        }) || this;
        _this.ipc = null;
        _this.ipc = require('electron').ipcRenderer;
        return _this;
    }
    SplashScreenPluginElectron.prototype.show = function (options, callback) {
        this.ipc.send('showCapacitorSplashScreen', tslib_1.__assign({}, options));
        if (callback) {
            callback();
        }
    };
    SplashScreenPluginElectron.prototype.hide = function (options, callback) {
        this.ipc.send('hideCapacitorSplashScreen', tslib_1.__assign({}, options));
        if (callback) {
            callback();
        }
    };
    return SplashScreenPluginElectron;
}(WebPlugin));
export { SplashScreenPluginElectron };
var SplashScreen = new SplashScreenPluginElectron();
export { SplashScreen };
