// @capacitor/electron/dist/esm/electron/toast.js
import * as tslib_1 from "tslib";
import { WebPlugin } from "@capacitor/core";
var ToastPluginElectron = /** @class */ (function (_super) {
    tslib_1.__extends(ToastPluginElectron, _super);
    function ToastPluginElectron() {
        var _this = _super.call(this, {
            name: 'Toast',
            platforms: ['electron']
        }) || this;
        _this.toastControllerPresent = true;
        var testel = document.querySelector('ion-toast-controller');
        if (testel == null) {
            console.error('No @ionic/core ToastController detected, please make sure you have included the web components.');
            _this.toastControllerPresent = false;
        }
        return _this;
    }
    ToastPluginElectron.prototype.show = function (options) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var toastController, duration, toastElement;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (!this.toastControllerPresent) return [3 /*break*/, 3];
                        toastController = document.querySelector('ion-toast-controller');
                        duration = 3000;
                        if (options.duration) {
                            duration = options.duration.localeCompare('long') === 0 ? 5000 : 3000;
                        }
                        return [4 /*yield*/, toastController.create({
                                position: 'bottom',
                                message: options.text,
                                duration: duration,
                            })];
                    case 1:
                        toastElement = _a.sent();
                        return [4 /*yield*/, toastElement.present()];
                    case 2: return [2 /*return*/, _a.sent()];
                    case 3: return [2 /*return*/, Promise.reject('No @ionic/core ToastController detected, please make sure you have included the web components.')];
                }
            });
        });
    };
    return ToastPluginElectron;
}(WebPlugin));
export { ToastPluginElectron };
var Toast = new ToastPluginElectron();
export { Toast };
